### PR TITLE
Mve2 us4 expand st1 var

### PR DESCRIPTION
**Pull request by HelloThere-ImMat**
Finished Var expansion

### CODE REVIEW

File: srcs/var/var_utils.c [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+/*   Created: 2023/03/31 16:59:54 by mat               #+#    #+#             */
+/*   Updated: 2023/04/03 17:50:20 by mat              ###   ########.fr       */
+/*                                                                            */
+/* ************************************************************************** */
+
+#include "minishell.h"
+
+bool	is_in_var_charset(const char c)
+{
+	return (ft_isalnum(c) == 1 || c == '_');
```
Comment (romann-broque) [inline srcs/var/var_utils.c:17]: '-' --> define

Comment (romann-broque) [inline srcs/var/var_utils.c:17]: ft_isalnum == true

File: srcs/var/handle_var.c [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+		"LAST_REV_VAL",
+		"minishell"
+	};
+	const size_t	index = abs_index(SPECIAL_VAR, c);
+
+	machine->line = replace_and_free(machine->line, array_str[index],
+			machine->i - 1, SPEC_VAR_LEN);
+}
+
+void	handle_var_start(t_vmachine *const machine)
```
Comment (romann-broque) [inline srcs/var/handle_var.c:28]: Special var state

File: srcs/var/handle_var.c [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+	{
+		machine->i++;
+		machine->word_len++;
+	}
+	else if (c != '\0' && is_in_str(SEPARATORS, c) == false)
+	{
+		if (is_special_var(c) == true)
+			replace_special_var(machine);
+		else
+			machine->line = replace_and_free(machine->line, "",
```
Comment (romann-broque) [inline srcs/var/handle_var.c:42]: #define EMPTY_STRING ""

Comment (romann-broque) [inline srcs/var/handle_var.c:42]: cut_string_at wrapper

File: srcs/var/handle_var.c [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+void	translate_var(t_vmachine *const machine)
+{
+	char	*var_name;
+	char	*var_value;
+
+	var_name = get_var_name(machine);
+	var_value = getenv(var_name);
+	free(var_name);
+	if (var_value == NULL)
+		machine->line = replace_and_free(machine->line, "",
```
Comment (romann-broque) [inline srcs/var/handle_var.c:69]: #define EMPTY_STRING ""

File: srcs/var/handle_var.c [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+/*                                                                            */
+/* ************************************************************************** */
+
+#include "minishell.h"
+
+static void	replace_special_var(t_vmachine *const machine)
+{
+	const char		c = machine->line[machine->i];
+	static char		array_str[][STR_LEN_MAX + 1] = {
+		"LAST_REV_VAL",
```
Comment (romann-broque) [inline srcs/var/handle_var.c:19]: #define

File: srcs/var/handle_var.c [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+		"minishell"
```
Comment (romann-broque) [inline srcs/var/handle_var.c:20]: #define

File: srcs/var/handle_var.c (hunk 2 above)
Comment (HelloThere-ImMat) [reply srcs/var/handle_var.c:28]: Faire un état specific pour le premier character de la variable

**Review by romann-broque — APPROVED** [review]

**Pull request merged**